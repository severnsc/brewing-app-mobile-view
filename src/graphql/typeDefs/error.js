const errorDef = `
    type Error {
        message: String!
        location: Location!
    }
`;

export default errorDef;
