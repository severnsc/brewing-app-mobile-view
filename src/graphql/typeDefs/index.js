import { makeExecutableSchema } from "graphql-tools";
import errorDef from "./error";
import userDef from "./user";
import locationDef from "./location";

const typeDefs = [userDef, errorDef, locationDef];

export default makeExecutableSchema({ typeDefs });
