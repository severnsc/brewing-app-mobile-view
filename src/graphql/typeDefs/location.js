const locationDef = `
    type Location {
        node: String!
        field: String!
    }
`;

export default locationDef;
