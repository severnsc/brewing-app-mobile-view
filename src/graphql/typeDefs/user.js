const userDef = `
    type User {
        id: String!
        username: String
        email: String,
        errors: [Error]
    }
`;

export default userDef;
