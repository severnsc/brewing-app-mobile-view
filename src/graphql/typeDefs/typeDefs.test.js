import typeDefs from ".";

describe("typeDefs", () => {
  it("matches snapshot", () => {
    expect(typeDefs).toMatchSnapshot();
  });
});
