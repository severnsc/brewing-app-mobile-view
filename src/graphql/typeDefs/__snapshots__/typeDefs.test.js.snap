// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`typeDefs matches snapshot 1`] = `
"
    type Error {
        message: String!
        location: Location!
    }


    type User {
        id: String
        username: String
        email: String
        errors: [Error]
    }

    input UserInput {
        username: String!
        email: String!
        password: String!
        confirmPassword: String!
    }

    input UserEdit {
        username: String
        email: String
        errors: [Error]
    }


    type Location {
        node: String!
        field: String!
    }
"
`;
